examples/astro-blog-demo/src/components/blog/CommentBlock.tsx

```diff
@@ -0,0 +1,201 @@
+              <div className="-mb-1 ms-4 h-4 w-5 self-end rounded-tl border-s-2 border-t-2 border-foreground/20"></div>
+
+              <ScrollAnchor targetHtmlId={parentComment._id}>
+                <Blog.Post.Comment.Status asChild>
+                  {({ status }) =>
+                    status === "DELETED" || status === "HIDDEN" ? (
+                      <span className="font-paragraph text-sm text-foreground/60">
+                        Replying to a deleted comment
+                      </span>
+                    ) : (
+                      <>
+                        <CommentAuthorAvatar size="sm" />
+                        <CommentContent
+                          asPlainText
+                          className="line-clamp-1 flex-1 font-paragraph text-sm text-foreground/60"
+                        />
+                      </>
+                    )
+                  }
+                </Blog.Post.Comment.Status>
+              </ScrollAnchor>
+            </div>
+          )}
+        </Comment.ParentComment>
+        <CommentAuthorAvatar />
+        <div className="grid flex-1 gap-y-2">
+          <Comment.Status asChild>
+            {({ status }) => {
+              switch (status) {
+                case "HIDDEN":
+                case "DELETED":
+                  return (
+                    <div className="grid min-h-8 items-center">
+                      <span className="font-paragraph text-foreground/60">
+                        This comment has been deleted
+                      </span>
+                    </div>
+                  );
+                default:
+                  return (
+                    <>
+                      <div className="-mt-1 flex items-center gap-2">
+                        <header className="grid flex-grow font-paragraph text-sm leading-normal">
+                          <div className="flex gap-2">
+                            <CommentAuthorName />
+                            {status === "PENDING" ? (
+                              <span>Awaiting approval</span>
+                            ) : null}
+                          </div>
+                          <CommentDate
+                            className="text-foreground/80"
+                            uiLocale={uiLocale}
+                          />
+                        </header>
+                        {isAuthenticated && isOwner ? (
+                          <DropdownMenu>
+                            <DropdownMenuTrigger asChild>
+                              <Button variant="ghost" size="icon">
+                                <EllipsisVerticalIcon className="h-4 w-4" />
+                              </Button>
+                            </DropdownMenuTrigger>
+                            <DropdownMenuPortal>
+                              <DropdownMenuContent>
+                                <DropdownMenuItem
+                                  onClick={() => {
```

Comment:
Use `Comment.Action.Delete` component for this, which wraps whatever you want it to, and pass it a callback `onDelete` which is not deleting when returns false